js/src/admin/components/UserListPage.tsx

```diff
@@ -0,0 +1,384 @@
+
+import type User from '../../common/models/User';
+
+import ItemList from '../../common/utils/ItemList';
+import classList from '../../common/utils/classList';
+import extractText from '../../common/utils/extractText';
+
+import AdminPage from './AdminPage';
+
+type ColumnData = {
+  /**
+   * Column title
+   */
+  name: String;
+  /**
+   * Component(s) to show for this column.
+   */
+  content: (user: User) => JSX.Element;
+};
+
+type ApiPayload = {
+  data: Record<string, unknown>[];
+  included: Record<string, unknown>[];
+  links: {
+    first: string;
+    next?: string;
+  };
+};
+
+type UsersApiResponse = User[] & { payload: ApiPayload };
+
+/**
+ * Admin page which displays a paginated list of all users on the forum.
+ */
+export default class UserListPage extends AdminPage {
+  /**
+   * Number of users to load per page.
+   */
+  private numPerPage: number = 50;
+
+  /**
+   * Current page number. Zero-indexed.
+   */
+  private pageNumber: number = 0;
+
+  /**
+   * Total number of forum users.
+   *
+   * Fetched from the active `AdminApplication` (`app`), with
+   * data provided by `AdminPayload.php`, or `flarum/statistics`
+   * if installed.
+   */
+  readonly userCount: number = app.data.modelStatistics.users.total;
+
+  /**
+   * Get total number of user pages.
+   */
+  private getTotalPageCount(): number {
+    if (this.userCount === -1) return 0;
+
+    return Math.ceil(this.userCount / this.numPerPage);
+  }
+
+  /**
+   * This page's array of users.
+   *
+   * `undefined` when page loads as no data has been fetched.
+   */
+  private pageData: User[] | undefined = undefined;
+
+  /**
+   * Are there more users available?
+   */
+  private moreData: boolean = false;
+
+  private isLoadingPage: boolean = false;
+
+  /**
+   * Component to render.
+   */
+  content() {
+    if (typeof this.pageData === 'undefined') {
+      this.loadPage(0);
+
+      return [
+        <section class="UserListPage-grid UserListPage-grid--loading">
+          <LoadingIndicator containerClassName="LoadingIndicator--block" size="large" />
+        </section>,
+      ];
+    }
+
+    const columns: (ColumnData & { itemName: string })[] = this.columns().toArray();
+
+    return [
+      <p class="UserListPage-totalUsers">{app.translator.trans('core.admin.users.total_users', { count: this.userCount })}</p>,
+      <section
+        class={classList(['UserListPage-grid', this.isLoadingPage ? 'UserListPage-grid--loadingPage' : 'UserListPage-grid--loaded'])}
+        style={{ '--columns': columns.length }}
+        role="table"
+        // +1 to account for header
+        aria-rowcount={this.pageData.length + 1}
+        aria-colcount={this.columns.length}
+        aria-live="polite"
+        aria-busy={this.isLoadingPage ? 'true' : 'false'}
+      >
+        {/* Render columns */}
+        {columns.map((column, colIndex) => (
+          <div class="UserListPage-grid--header" role="columnheader" aria-colindex={colIndex + 1} aria-rowindex={1}>
```

Comment:
also, this should be `UserListPage-grid-header` since it's not a modifier but a descendant.